lib/models/metrics.js

```diff
@@ -0,0 +1,197 @@
+ * please contact with::[contacto@tid.es]
+ */
+
+'use strict';
+
+var logger = require('logops'),
+    metrics = {};
+
+var metricsNames = [
+    'incomingTransactions',
+    'incomingTransactionsRequestSize',
+    'incomingTransactionsResponseSize',
+    'incomingTransactionsErrors',
+
+    'serviceTime',
+
+    'outgoingTransactions',
+    'outgoingTransactionsRequestSize',
+    'outgoingTransactionsResponseSize',
+    'outgoingTransactionsErrors',
+
+    'notifications',
+    'okNotifications',
+    'failedNotifications',
+
+    'ruleCreation',
+    'okRuleCreation',
+    'failedRuleCreation',
+
+    'ruleUpdate',
+    'okRuleUpdate',
+    'failedRuleUpdate',
+
+    'ruleDelete',
+    'okRuleDelete',
+    'failedRuleDelete',
+
+    'firedRules',
+    'failedAction',
+
+    'actionEntityUpdate',
+    'okActionEntityUpdate',
+    'failedActionEntityUpdate',
+
+    'actionSMS',
+    'okActionSMS',
+    'failedActionSMS',
+
+    'actionEmail',
+    'okActionEmail',
+    'failedActionEmail',
+
+    'actionHttpPost',
+    'okActionHttpPost',
+    'failedActionHttpPost',
+
+    'actionTwitter',
+    'okActionTwitter',
+    'failedActionTwitter'
+];
+
+function newMetrics() {
+    var m = {};
+    metricsNames.forEach(function(name) {
+        m[name] = 0;
+    });
+    return m;
+}
+
+function getMetrics(service, servicePath) {
+    var mService,
+        mServicePath;
+
+    mService = metrics[service];
+    if (!mService) {
+        mService = metrics[service] = {};
+    }
+    mServicePath = mService[servicePath];
+    if (!mServicePath) {
+        mServicePath = mService[servicePath] = newMetrics();
+    }
+
+    return mServicePath;
+}
+
+function deleteMetrics() {
+    metrics = {};
+}
+
+function incMetrics(service, servicePath, key, value) {
+    var m = getMetrics(service, servicePath);
+    if (value === undefined) {
+        value = 1;
+    }
+    if (m[key] === undefined) {
+        logger.error('invalid key for metrics %s', key);
+        return;
+    }
+    m[key] += value;
+}
+
+function decoratedMetrics() {
+    var decorated = {services: {}},
+        total = newMetrics(),
+        totalPerServ;
+
+    for (var serv in metrics) {
+        totalPerServ = newMetrics();
+        decorated.services[serv] = {subservices: {}};
+        for (var subserv in metrics[serv]) {
+            for (var key in metrics[serv][subserv]) {
+                totalPerServ[key] += metrics[serv][subserv][key];
+                total[key] += metrics[serv][subserv][key];
+            }
+            decorated.services[serv].subservices[subserv] = metrics[serv][subserv];
+        }
+        decorated.services[serv].sum = totalPerServ;
+    }
+    decorated.sum = total;
+    return decorated;
+}
+
+module.exports.GetMetrics = getMetrics;
+module.exports.DeleteMetrics = deleteMetrics;
+module.exports.IncMetrics = incMetrics;
+module.exports.GetDecorated = decoratedMetrics;
+
+module.exports.incomingTransactions = 'incomingTransactions';
```

Comment:
Also not easy to simplify the use in the array ... we could en writing the same name 3 times ...

I'd left as it is just in this case